BML_TUX_Project/Packages/bmlTUX/Scripts/ExperimentParts/Session.cs

```diff
@@ -33,39 +40,39 @@ public class Session {
         }
 
 
-        public static Session LoadSessionData() {
-            string filePath = FileLocationSettings.LastSessionSaveFilePath;
+        public static Session LoadSessionData(FileLocationSettings fileLocationSettings) {
             Session session;
+            string filePath = fileLocationSettings.SessionSaveFilePath;
             if (File.Exists(filePath)) {
                 string dataAsJason = File.ReadAllText(filePath);
-                try {
+                try{
                     session = JsonUtility.FromJson<Session>(dataAsJason);
-                }
-                catch (ArgumentException) {
+                }catch (ArgumentException){
                     File.Delete(filePath);
-                    Debug.LogWarning($"{TuxLog.Prefix} Previous session file became corrupt, deleting it.");
-                    session = CreateNewSession();
+                    Debug.Log($"{TuxLog.Prefix} Previous Session file corrupt, deleting");
+                    session = CreateNewSession(fileLocationSettings);
                 }
                 
             }
             else {        
-                session = CreateNewSession();
+                session = CreateNewSession(fileLocationSettings);
             }
+
             session.Enable();
             return session;
         }
 
-        static Session CreateNewSession() {
-            Session session;
+        static Session CreateNewSession(FileLocationSettings fileLocationSettings) {
+            Session session = new Session();
             Debug.Log($"{TuxLog.Prefix} Previous Session file not found, creating new");
-            session = new Session();
+            session.saveFilePath = fileLocationSettings.SessionSaveFilePath;
+            session.sessionFolder = fileLocationSettings.SessionFolder;
```

Comment:
I don't love this. Because it could introduce bugs when future changes are made if we forget to set one of these fields. Suggest passing the FileLocationSettings reference into the Session constructor.

Reply:
Changed in: 3eb251e42350fa91cf4b8307a77c87236d1cec98